daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -1584,35 +1584,35 @@ object Main extends Logging {
     } catch {
       case s: scala.util.control.ControlThrowable => throw s
       case e: java.io.FileNotFoundException => {
-        log(LogLevel.Error, "%s", e.getMessage())
+        log(LogLevel.Error, "Missing File Error", e.getMessage())
         1
       }
       case e: InvalidParserException => {
-        log(LogLevel.Error, "%s", e.getMessage())
-        1
+        log(LogLevel.Error, "Invalid Parser Error", e.getMessage())
+        2
       }
       case e: ExternalVariableException => {
-        log(LogLevel.Error, "%s", e.getMessage())
-        1
+        log(LogLevel.Error, "External Variable Error", e.getMessage())
+        3
       }
       case e: BindingException => {
-        log(LogLevel.Error, "%s", e.getMessage())
-        1
+        log(LogLevel.Error, "Binding Error", e.getMessage())
+        4
       }
       case e: NotYetImplementedException => {
         nyiFound(e)
       }
       case e: TDMLException => {
-        log(LogLevel.Error, "%s", e.getMessage())
-        1
+        log(LogLevel.Error, "TDML Error", e.getMessage())
+        5
       }
       case e: OutOfMemoryError => {
         oomError(e)
       }
       case e: UserDefinedFunctionFatalErrorException => {
-        log(LogLevel.Error, "%s", e.getMessage())
+        log(LogLevel.Error, "User Error", e.getMessage())
         e.printStackTrace()
-        1
+        6
```

Comment:
Instead of having all these magic numbers spread out, I would suggest we create some vals so that these are self documenting. e.g. val USER_DEFINED_FUNCTION_ERROR_CODE. That way all the codes can be define in one spot and used throughout.